pkg/noderesourcetopology/match.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/klog/v2"
+	v1qos "k8s.io/kubernetes/pkg/apis/core/v1/helper/qos"
+	bm "k8s.io/kubernetes/pkg/kubelet/cm/topologymanager/bitmask"
+	framework "k8s.io/kubernetes/pkg/scheduler/framework/v1alpha1"
+	apiconfig "sigs.k8s.io/scheduler-plugins/pkg/apis/config"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	topoclientset "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/clientset/versioned"
+	topoinformerexternal "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	topologyinformers "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions"
+	topoinformerv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/generated/informers/externalversions/topology/v1alpha1"
+)
+
+const (
+	// NodeResourceTopologyMatchName is the name of the plugin used in the plugin registry and configurations.
+	NodeResourceTopologyMatchName = "NodeResourceTopologyMatch"
+)
+
+var _ framework.FilterPlugin = &NodeResourceTopologyMatch{}
+
+type nodeTopologyMap map[string]topologyv1alpha1.NodeResourceTopology
+
+type PolicyHandler interface {
+	PolicyFilter(pod *v1.Pod, zoneMap topologyv1alpha1.ZoneList) *framework.Status
+}
+
+type PolicyHandlerMap map[apiconfig.TopologyManagerPolicy]PolicyHandler
+
+// NodeResourceTopologyMatch plugin which run simplified version of TopologyManager's admit handler
+type NodeResourceTopologyMatch struct {
+	handle framework.FrameworkHandle
+
+	nodeTopologyInformer    topoinformerv1alpha1.NodeResourceTopologyInformer
+	topologyInformerFactory topoinformerexternal.SharedInformerFactory
+
+	nodeTopologies         nodeTopologyMap
+	nodeTopologyGuard      sync.RWMutex
+	topologyPolicyHandlers PolicyHandlerMap
+}
+
+type SingleNUMANodeHandler struct {
+	match *NodeResourceTopologyMatch
+}
+
+type PodLevelResourceHandler struct {
+	match *NodeResourceTopologyMatch
+}
```

Comment:
agree, now I reworked this part and removed match * field, but 2 structures still necessary because of 2 different behavior, we discussed it in another comments